config/envoyconfig/listeners.go

```diff
@@ -853,7 +853,7 @@ func hostMatchesDomain(u *url.URL, host string) bool {
 func getPoliciesForDomain(options *config.Options, domain string) []config.Policy {
 	var policies []config.Policy
 	for _, p := range options.GetAllPolicies() {
-		if p.Source != nil && hostMatchesDomain(p.Source.URL, domain) {
```

Comment:
The `hostMatchesDomain` function was intended to be used with a URL and a host, not a domain (e.g. `example.com:1234` not `example.com`). Confusingly the `Domain` in the name refers to how the matching is done not that it expects a domain.